scene/3d/skeleton_3d.h

```diff
@@ -186,6 +187,10 @@ class Skeleton3D : public Node3D {
 
 	void clear_bones();
 
+	// these functions are for the editor gizmo
+	Bone get_bone_raw_struct(int p_bone);
+	void set_bone_raw_struct(int p_bone, Bone p_bone_struct);
+
```

Comment:
Are these being used in this pr directly?

Reply:
You're free to add it your subgizmos pr though.

Reply:
Exposing would require the struct to be a resource with all the wrappers and gdscript bindings.

Reply:
> Exposing would require the struct to be a resource with all the wrappers and gdscript bindings.

I know, I went down that route. Adding the custom resource with all the wrappers and GDScript bindings was 3-4 of the squished commits! I ultimately found it was not worth the extra complexity and using the Struct was more clunky than just using the functions in Skeleton.

I can probably remove those functions and move the Bone struct back to a private struct now. As you said, if the sub-gizmo needs it, it can be changed again. I'll make a PR with these changes soon.

Reply:
The last pr should be combined with the pr that made the change so it has a change commit. Note that history is lost here.

Reply:
I tried, but the rebase gets all messed up when I try to move the commit to override the first commit. Is there a way to rebase and reorder the commits without the log getting messed up? I tried moving the actual `fixup`/`squash` line so it was under the initial commit, and it didn't work. There was merge conflicts for every single commit, even for those that would otherwise be unaffected.

Reply:
It's tricky, and that's why I squish all of my commits together, because it makes porting to 3.2 easier. (One cherry pick and then fix conflicts.)